model/jpa/src/main/java/org/keycloak/authorization/jpa/store/JPAPolicyStore.java

```diff
@@ -269,14 +291,13 @@ public Policy findByName(String name, String resourceServerId) {
         query.setParameter("serverId", resourceServerId);
 
         List<String> result = query.getResultList();
-        List<Policy> list = new LinkedList<>();
+
         for (String id : result) {
             Policy policy = provider.getStoreFactory().getPolicyStore().findById(id, resourceServerId);
             if (Objects.nonNull(policy)) {
-                list.add(policy);
+                consumer.accept(policy);
```

Comment:
`for` cycle could be replaced by
```java
result.stream()
  .map(id -> provider.getStoreFactory().getPolicyStore().findById(id, resourceServerId))
  .filter(Objects::nonNull)
  .forEach(consumer::accept);
```
which would prepare it to [JPA streams](https://vladmihalcea.com/whats-new-in-jpa-2-2-stream-the-result-of-a-query-execution/)